pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go

```diff
@@ -358,3 +328,180 @@ func TestRestart(t *testing.T) {
 		close(shutdownChan)
 	}
 }
+
+func Test_migrateConfig(t *testing.T) {
+	type shutdownConfig struct {
+		shutdownGracePeriodRequested    time.Duration
+		shutdownGracePeriodCriticalPods time.Duration
```

Comment:
i may be missing this elsewhere, but it seems that we could get a negative value if graceperiodcriticalpods is greater than graceperiodrequested.

Reply:
Kubelet configuration validation has the check that ensures 

```
kc.ShutdownGracePeriodCriticalPods.Duration <= kc.ShutdownGracePeriod.Duration
```


https://github.com/kubernetes/kubernetes/blob/e67979eaf6e3ae9b8be67ae8374a3e008f88db76/pkg/kubelet/apis/config/validation/validation.go#L145-L147